contracts/rewards/src/contract/execute.rs

```diff
@@ -192,12 +200,14 @@ where
         // (i.e. we are in epoch 0, which started at block 0 and epoch duration is 1000. At epoch 500, the params
         // are updated to shorten the epoch duration to 100 blocks. We set the epoch number to 1, to prevent skipping
         // epochs 1-4, and so all events prior to the start of epoch 1 have an epoch number of 0)
-        let should_end =
-            cur_epoch.block_height_started + u64::from(new_params.epoch_duration) < block_height;
+        let should_end = cur_epoch
+            .block_height_started
+            .saturating_add(u64::from(new_params.epoch_duration))
```

Comment:
while the event is highly unlikely, if it happens the current epoch is suddenly shorter than what we expect based on the epoch duration. I don't think we should break that invariant, so I'd prefer a `checked_add()?` here